<?xml version="1.0"?>
<launch>
    
    <arg name="camera_frame" default="realsense_optical" />
    <arg name="camRaw" default="/realsense_rgb/image_raw" doc="image_raw topic"/>
    <arg name="camInfo" default="/realsense_rgb/camera_info" doc="camera_info topic"/>
    
    <arg name="idBits" default="5" doc="Number of encoded bits"/>
    <arg name="idSamples" default="360" doc="Number of samples to detect ID"/>
    <arg name="hammingDist" default="1" doc="Hamming distance used for ID"/>
    <arg name="maxMarkers" default="10" doc="Maximum possible amount of markers to be tracked"/>
    <arg name="useGui" default="true" doc="Whether to use GUI"/>
    <arg name="useAcuteFilter" default="true" doc="only whycons within an acute angle repect the camera will be published"/>
    <arg name="maxDetectionDistance" default="6" doc=" whycon futher away than this distance won't be published"/>s
    <arg name="minDetectionsToPublish" default="2" doc="minimum amount of detected fiducial before start publishing"/>
    <arg name="fontPath" value="$(find whycon_ros)/etc/font.ttf"/>
    <arg name="calibDefPath" value="$(find whycon_ros)/etc/default.cal"/>

    <node pkg="whycon_ros" type="whycon_ros" name="whycon_ros" respawn="true" output="screen" args="$(arg fontPath) $(arg calibDefPath)" launch-prefix="">
      <remap from="/camera/camera_info" to="$(arg camInfo)"/>
      <remap from="/camera/image_raw" to="$(arg camRaw)"/>
      <param name="circleDiameter" value="40"/>
      <param name="numMarkers" value="28"/>
      <param name="useGui" value="$(arg useGui)"/>
      <param name="idBits" value="$(arg idBits)"/>
      <param name="idSamples" value="$(arg idSamples)"/>
      <param name="hammingDist" value="$(arg hammingDist)"/>
      <param name="maxMarkers" value="$(arg maxMarkers)"/>
      <param name="useAcuteFilter" value="$(arg useAcuteFilter)"/>
      <param name="maxDetectionDistance" value="$(arg maxDetectionDistance)"/>
      <param name="minDetectionsToPublish" value="$(arg minDetectionsToPublish)"/>
    </node>
    
</launch>

